enhancements/installer/multi-disk-setup.md

```diff
@@ -0,0 +1,265 @@
+
+## Summary
+
+This enhancement proposes to extend the installer to support configurations for multiple disks. This will allow users to define specific roles for additional disks, such as for `etcd` data, `swap` space, or other `user-defined` mount points. The installer will generate the necessary ignition configuration to partition, format, and mount these disks as specified by the user during the installation process.
+
+## Motivation
+
+Currently, the installer has limited explicit support for configuring multiple disks with distinct roles during the initial setup. Users often require dedicated disks for performance, capacity, or specific application needs (e.g., separating etcd I/O, providing swap). This enhancement aims to provide a streamlined and supported way to declare these multi-disk configurations directly through the installer, simplifying the deployment process for such scenarios.
+
+### User Stories
+
+* As a cluster administrator, I want to specify a dedicated disk for etcd during installation, so that I can ensure etcd has isolated I/O performance and dedicated storage.
+* As a cluster administrator, I want to configure a dedicated swap disk during installation, so that I can provide swap space for nodes that require it.
+* As a cluster administrator, I want to define custom mount points on additional disks for specific application data or utilities during installation, so that I can prepare nodes with pre-configured storage layouts.
+* As an installer developer, I want a clear API within the machine pool definition to specify additional disks and their purposes, so that I can reliably generate the corresponding ignition configurations.
+
+### Goals
+
+* Enable the installer to accept configurations for multiple disks per machine pool.
+* Define clear types for these additional disks (e.g., Etcd, Swap, UserDefined).
+* Generate ignition configurations that create the necessary partitions, file systems (e.g., XFS, ext4), and systemd mount units for the specified disks.
+* Ensure the initial implementation is isolated to the installer and its ignition generation capabilities.
+
+### Non-Goals
+
+* Dynamic disk management post-installation (this will be handled by storage operators or manual intervention).
+* Complex RAID configurations via the installer (simple disk partitioning and formatting is the focus).
+* Support for all possible file systems; a common default (e.g., XFS or ext4) will be used, with potential for future extension.
+* Changes to how the operating system image itself is deployed to the primary disk.
+
+## Proposal
+
+The core of this proposal is to introduce new structures within the machine pool API that allow users to declare additional disks and their intended use. The installer will then interpret these declarations to generate the appropriate `storage` and `systemd` sections in the ignition config.
```

Comment:
Just to clarify this point, do these APIs propagate to the cluster post-install? Or is it just used to generate installer-specific ignition data.

Another way to put it would be whether the user has any say in modifying this configuration post-install via any  API field or would they have to manually modify user-data secrets, since I presume that it will end up on the cluster's machine-api namespaced secrets